crypto/rand/drbg_lib.c

```diff
@@ -33,9 +28,79 @@ static const char ossl_pers_string[] = "OpenSSL NIST SP 800-90A DRBG";
  * a much bigger deal than just re-setting an allocated resource.)
  */
 
+/*
+ * The three shared DRBGs
+ *
+ * There are three shared DRBGs (master, public and private), which are
+ * accessed concurrently by all threads.
+ *
+ * The master DRBG
+ *
+ * Not used directly by the application, only for reseeding the two other
+ * DRBGs. It reseeds itself by pulling either randomness from os entropy
+ * sources or by consuming randomnes which was added by RAND_add()
+ */
+static RAND_DRBG drbg_master;
+/*
+ * The public DRBG
+ *
+ * Used by default for generating random bytes using RAND_bytes().
+ */
+static RAND_DRBG drbg_public;
+/*
+ * The private DRBG
+ *
+ * Used by default for generating private keys using RAND_priv_bytes()
+ */
+static RAND_DRBG drbg_private;
+/*
+ * In addition there are DRBGs, which are not shared, but used only by a
+ * single thread at every time, for example the DRBGs which are owned by
+ * an SSL context.
+ *
+ * This gives the following overall picture:
+ *
+ *                  <os entropy sources>
+ *                         |
+ *                         v
+ *    RAND_add() ==>    <master>          \
+ *                       /   \            | shared DRBGs
+ *                      v     v           |  (with locking)
+ *                 <public>  <private>    /
```

Comment:
In the long term, this documentation should be moved to a proper place in the manual pages. How about adding a drbg.pod to man7 explaining all the general concepts, in addition to adding the RAND_DRBG API documentation in man3? (This will be done in a separate PR which publishes the API).

Reply:
>  How about adding a drbg.pod to man7 explaining all the general concepts

This manual page would not only contain this comment, but a general overview over the entire new DRBG API, with some general concepts explained and references to the NIST document.

Reply:
Great.  Please make all special-formatted comments start with "/*+" so that a future indent run, if/when it happens, doesn't mangle the picture and such.